File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
var maxProfit = function(prices) {
    var buy = prices[0]
    var profit = 0
    for(let i=0;i<prices.length;i++){
        y = prices[i]-buy
        if(buy > prices[i]){
            buy = prices[i]
        }
        else{
            profit = Math.max(profit,y)
        }
    }
    return profit
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # buy = prices[0]
        # sell = 0
        # profit = 0
        # for i in range(1,len(prices)):
        #     sell = prices[i]
        #     y = sell - buy
        #     if buy > sell:
        #         buy = sell
        #     else:
        #         profit = max(profit,y)    
        # return profit  

        profit = 0
        buy = prices[0]
        for i in range(1,len(prices)):
            y = prices[i]-buy
            if buy > prices[i]:
                buy = prices[i]
            else:
                profit = max(profit,y)
        return profit
